src/ape/api/address.py

```diff
@@ -22,15 +35,32 @@ def provider(self) -> ProviderAPI:
 
     @provider.setter
     def provider(self, value: ProviderAPI):
+        """
+        Sets the active provider if connected to one.
+
+        Args:
+            value (:class:`~ape.api.providers.ProviderAPI`): The provider to set.
+        """
+
         self._provider = value
 
     @property
     @abstractmethod
     def address(self) -> AddressType:
-        ...
+        """
+        The raw address type.
+
+        Returns:
+            :class:`~ape.types.AddressType`
+        """
 
     def __dir__(self) -> List[str]:
-        # This displays methods to IPython on `a.[TAB]` tab completion
+        """
+        Display methods to IPython on ``a.[TAB]`` tab completion.
+
+        Returns:
+            list[str]: Method names that IPython uses for tab completion.
+        """
         return [
             "address",
             "balance",
```

Comment:
tbh would like to add better integration with this via the base class, but can only do that once pydantic upgrade happens

Reply:
You mean the feature for IPython or the docs for it?
If the feature, we should not add docs for this.

Reply:
btw, I have started looking into pydantic work and it is on my list! I have rebased the branch at least.

Reply:
> You mean the feature for IPython or the docs for it? If the feature, we should not add docs for this.

oh I mean the feature of integrating with IPython, but looking at methods and fields of the dataclass

Reply:
Okay cool. Well if we delete this method / approach, that is no big deal.